src/utils.typ

```diff
@@ -802,6 +802,56 @@
   )
 }
 
+// recursively checks if `it` has a text in it
+#let _contains-text(it) = {
+  if type(it) != content {
+    return false
+  }
+  if it.func() == text {
+    return true
+  }
+  if it.has("body") {
+    return _contains-text(it.body)
+  }
+  if it.has("base") {
+    return _contains-text(it.base)
+  }
+  if it.has("child") {
+    return _contains-text(it.child)
+  }
+  if it.has("children") {
```

Comment:
The `table` and `grid` types also have `children`, I'm not sure if this should be included?

Reply:
I understand. It might be good to have an option to toggle the behaviour, or exclude those elements (with using `util.is-sequence`?)

Reply:
Since the table has borders, we should use `util.is-sequence` as default, but it can indeed be made into a configuration.